packages/entities/entities-plugins/src/components/PluginIcon.vue

```diff
@@ -31,11 +30,27 @@ defineProps({
   },
 })
 
+const defaultIcon = ref('')
+const src = ref('')
 const img = ref<HTMLImageElement>()
 
-const onError = () => {
-  if (img.value) {
-    img.value.src = defaultIcon
+const onError = (): void => {
+  if (img.value && defaultIcon.value) {
+    img.value.src = defaultIcon.value
   }
 }
+
+onBeforeMount(() => {
+  try {
+    defaultIcon.value = new URL('../assets/images/plugin-icons/missing.png', import.meta.url).href
+  } catch (err) {
+    defaultIcon.value = ''
+  }
+
+  try {
+    src.value = composables.getPluginIconURL(props.name)
+  } catch (err) {
+    src.value = defaultIcon.value
+  }
```

Comment:
question: why isn't the try/catch in the composable where the import is actually happening?